src/components/containers/victory-zoom-container.js

```diff
@@ -131,12 +132,23 @@ export const zoomContainerMixin = (base) => class VictoryZoomContainer extends b
     const childComponents = React.Children.toArray(props.children);
 
     return childComponents.map((child) => {
-      const { currentDomain } = props;
+      const { currentDomain, zoomActive } = props;
       const originalDomain = defaults({}, props.originalDomain, props.domain);
       const zoomDomain = defaults({}, props.zoomDomain, props.domain);
       const cachedZoomDomain = defaults({}, props.cachedZoomDomain, props.domain);
-      const domain = isEqual(zoomDomain, cachedZoomDomain) ?
-        defaults({}, currentDomain, originalDomain) : zoomDomain;
+
+      let domain;
+      if (!isEqual(zoomDomain, cachedZoomDomain)) {
+        // if zoomDomain has been changed, use it
+        domain = zoomDomain;
+      } else if (!zoomActive) {
+        // if user has zoomed all the way out, use the child domain
+        domain = child.props.domain;
```

Comment:
`child.props.domain` is already calculated for us, so this approach has no new computational overhead.